### PR TITLE
Clean Up: move pubkey parsing to config

### PR DESCRIPTION
**Pull request by JoeGruffins**
part of issue #226 

Move validation of pubkeys, length and network, to config.

I made the parsed keys into package level variables because it felt weird to include them in the config struct since they are not configured by the user.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -461,6 +495,13 @@ func loadConfig() (*config, []string, error) {
 		}
 	}
 
+	if err := cfg.parsePubKeys(activeNetParams.Params); err != nil {
+		str := "%s: Failed to parse Pubkeys"
+		err := fmt.Errorf(str+": %v", funcName, err)
```
Comment (jholdstock) [inline config.go]: Please avoid string concatenation here. You can instead use

```
err := fmt.Errorf("%s: Failed to parse Pubkeys: %v", funcName, err)
```

Comment (JoeGruffins) [reply config.go]: Yes sir!

I also realized the check should be made after checking whether keys are set at all, so moved it down a bit.

File: config.go [hunk 2]
```diff
@@ -259,6 +263,36 @@ func fileExists(name string) bool {
...
+	if err != nil {
+		return fmt.Errorf("voting wallet extended public key: %v", err)
+	}
+	if !votingWalletVoteKey.IsForNet(params) {
+		return fmt.Errorf("voting wallet extended public key is for wrong network")
+	}
+	return nil
+}
+
+func (c *config) GetFeeKey() *hdkeychain.ExtendedKey {
```
Comment (chappjc) [inline config.go]: Style issue: https://golang.org/doc/effective_go.html#Getters
We try to follow this in general

Comment (chappjc) [reply config.go]: But these getters seem unneeded.  `coldWalletFeeKey` is already accessible by main (same package).  Also, the receiver of this getter is pointless.

Comment (JoeGruffins) [reply config.go]: Thank you Chapp! I will start reading through effective go.

I didn't even realise they were the same package.  Still learning, thank you.

Is this better?

**Comment by chappjc** [comment]
OK by me.  The new `parsePubKeys` function is a candidate for a test case, but I won't make a fuss about it as long as testing on a testnet vsp/stakepool setup has been done.  @JoeGruffins have you put it through the paces on a full setup?

**Comment by JoeGruffins** [comment]
@chappjc 
I checked several times to make sure the keys were being sent to controllers/main.go properly.  I also ran my testnet node for a day with this code and ticket purchases and fees were unaffected.  

But I want to write a test! You mean the testing package right? I want to do that. Give me a day.

**Comment by chappjc** [comment]
Great!  Yup, some test in a new config_test.go file.

**Comment by JoeGruffins** [comment]
test added

**Pull request merged**